src/ray/util/file_persistence.h

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <chrono>
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <thread>
+
+#include "nlohmann/json.hpp"
+#include "ray/common/id.h"
+#include "ray/common/status.h"
+#include "ray/common/status_or.h"
+
+namespace ray {
+
+/// Write a key-value pair to a file atomically.
+/// Uses temp file + rename for cross-platform(linux/windows and
+/// cross-language(C++/Python) safe file sharing.
+inline Status WriteKVFile(const std::string &file_path,
```

Comment:
nit: So a NEW pattern we've been doing is trying to use StatusOr.  Can you take a look at that type and see if we can adapt it for this new interface?

Reply:
oh nevermind, you use StatusOr later and actually status makes more sense since there's no 'Or' here.  But one thing I'd like to see, can you please enumerate the status codes in the description and utilize Doxygen comment format for documeting all of these?

This definitely isn't the pattern through the code base today, but it's a standard we're trying to champion.  Take a look at any of the cgroup interfaces or actor_info_accessor_interface.h for examples of this.